### PR TITLE
Add configurable poll interval for Hue bridge.

### PR DESCRIPTION
**Pull request by scottydj2**
Use Mode1 hardware table entry to store the poll interval.
If 0, default of 10s is used.

### CODE REVIEW

File: hardware/PhilipsHue/PhilipsHue.cpp [hunk 1]
```diff
@@ -48,12 +48,13 @@ string ReadFile(string filename)
...
 #endif
 
-CPhilipsHue::CPhilipsHue(const int ID, const string &IPAddress, const unsigned short Port, const string &Username) :
+CPhilipsHue::CPhilipsHue(const int ID, const string &IPAddress, const unsigned short Port, const string &Username, const int poll) :
 m_IPAddress(IPAddress),
 m_UserName(Username)
 {
 	m_HwdID=ID;
 	m_Port = Port;
+	m_poll = poll;
```
Comment (gizmocuz) [inline hardware/PhilipsHue/PhilipsHue.cpp]: Please call it m_poll_interval

File: hardware/PhilipsHue/PhilipsHue.cpp [hunk 2]
```diff
@@ -93,10 +94,23 @@ bool CPhilipsHue::StopHardware()
 void CPhilipsHue::Do_Work()
 {
 	int msec_counter = 0;
-	int sec_counter = HUE_POLL_INTERVAL-2;
+	int sec_counter = 0;
 
 	_log.Log(LOG_STATUS,"Philips Hue: Worker started...");
 
+	// Catch uninitialised Mode1 entry.
+	if (m_poll < 1)
```
Comment (gizmocuz) [inline hardware/PhilipsHue/PhilipsHue.cpp]: Should be checked at constructor of the claxx

**Comment by scottydj2** [comment]
Updates made as requested.

File: hardware/PhilipsHue/PhilipsHue.cpp [hunk 3]
```diff
@@ -88,15 +103,16 @@ bool CPhilipsHue::StopHardware()
     return true;
 }
 
-#define HUE_POLL_INTERVAL 10
 
 void CPhilipsHue::Do_Work()
 {
 	int msec_counter = 0;
-	int sec_counter = HUE_POLL_INTERVAL-2;
+	int sec_counter = 0;
```
Comment (gizmocuz) [inline hardware/PhilipsHue/PhilipsHue.cpp]: i would make this

int sec_counter = m_poll_interval - 1;

and remove the below line after the log message

**Comment by gizmocuz** [comment]
Thanks!

**Pull request merged**